sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -53,62 +58,110 @@ object Klaviyo {
      * (e.g. after a fresh login)
      *
      * @param phoneNumber Phone number for active user
+     * @return
      */
     fun setPhoneNumber(phoneNumber: String): Klaviyo =
-        this.setProfile(Profile().setPhoneNumber(phoneNumber))
+        this.setProfileAttribute(ProfileKey.PHONE_NUMBER, phoneNumber)
+
+    /**
+     * @return The phone number of the currently tracked profile, if set
+     */
+    fun getPhoneNumber(): String? = UserInfo.phoneNumber.ifEmpty { null }
```

Comment:
Is there a kotlin way to expose a var and just override the setter getter? Might be less code…

Reply:
Yes, but I feel that keeping setters as functions is more indicative that there is work going on to sync with the API. Also methods are chain-able, assignments would not be (and I'm just a big fan of chaining)
```
Klaviyo.email = "test@gmail.com" // Seems like a simple assignment, obscures the fact that it is going to produce a side effect
```